### ISSUE TITLE
Fix S4136 FP: should not raise when static methods are grouped together

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#2776]
### Description

[from community](https://community.sonarsource.com/t/false-positive-for-s4136/15787)

If we group static methods together, it should not be a code smell.

Currently, it raises an issue - see below

### Repro steps

```
    public class StaticMethodsTogether
    {
        public static void MethodA() // Noncompliant FP - static methods are grouped together, it's ok
        {
            // do nothing
        }

        public static void MethodB()
        {
            // do nothing
        }

        public void MethodA(int i)
        {
            // do nothing
        }
    }
```

### Known workarounds

Group static and non-static overloads together

```
    public class StaticMethodsTogether
    {
        public static void MethodA()
        {
            // do nothing
        }
        public void MethodA(int i)
        {
            // do nothing
        }
        public static void MethodB()
        {
            // do nothing
        }


    }
```

### Related information

* SonarC# Version 8.0

**Comment by andrei-epure-sonarsource** [comment]
this happens for VB . NET as well

**Comment by pavel-mikula-sonarsource** [comment]
Same issue applies to abstract/MustOverride members as well.
